src/main/java/hudson/plugins/warnings/parser/CadenceIncisiveParser.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
+package hudson.plugins.warnings.parser;
+import java.util.regex.Matcher;
+import hudson.Extension;
+import hudson.plugins.analysis.util.model.Priority;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ *
+ * A parser for Cadence Incisive Enterprise Simulator
+ * @author Andrew 'Necromant' Andrianov
+ */
+
+@Extension
+public class CadenceIncisiveParser extends RegexpLineParser {
+    private static final String SLASH = "/";
+    private static final String CADENCE_MESSAGE_PATTERN = "("
+            + "(^[a-zA-Z]+): \\*([a-zA-Z]),([a-zA-Z]+): (.*) \\[File:(.*), Line:(.*)\\]." //ncelab vhdl warning
+             + ")|("
+            + "(^[a-zA-Z]+): \\*([a-zA-Z]),([a-zA-Z]+) \\((.*),([0-9]+)\\|([0-9]+)\\): (.*)$" //Warning/error with filename
+            + ")|("
+            + "(^g?make\\[.*\\]: Entering directory)\\s*(['`]((.*))\\')" // make: entering directory
+            + ")|("
+            + "(^[a-zA-Z]+): \\*([a-zA-Z]),([a-zA-Z]+): (.*)$" //Single generic warning
+            + ")";
+    private String directory = "";
+    
+    /**
+     * Creates a new instance of {@link CadenceIncisiveParser}.
+     */
+    public CadenceIncisiveParser() {
+        super(Messages._Warnings_CadenceIncisive_ParserName(),
+                Messages._Warnings_CadenceIncisive_LinkName(),
+                Messages._Warnings_CadenceIncisive_TrendName(),
+                CADENCE_MESSAGE_PATTERN);
+    }
+
+    @Override
+    protected String getId() {
+        return "Cadence Incisive Enterprise Simulator";
+    }
+
+    
+    private Warning handleDirectory(final Matcher matcher, int offset) {
+        directory = matcher.group(offset) + SLASH; //17
+        return FALSE_POSITIVE;
+    }
+
+    
+    @Override
+    protected Warning createWarning(final Matcher matcher) {
+
+        String tool, type, category, message, fileName; 
```

Comment:
Single line per variable. Also move variables to the position where they are used first.